actors/builtin/miner/miner_actor.go

```diff
@@ -508,6 +541,24 @@ func (a Actor) ConfirmSectorProofsValid(rt Runtime, params *builtin.ConfirmSecto
 			VerifiedDealWeight: dealWeights.VerifiedDealWeight,
 		}
 
+		replaceSector := precommit.Info.ReplaceSector
+		if replaceSector != 0 {
+			info, exists, err := st.GetSector(store, replaceSector)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to check sector %v", replaceSector)
+			faulty, err := st.Faults.IsSet(uint64(replaceSector))
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to check sector %v fault state", replaceSector)
```

Comment:
We should check this at ProveCommit to avoid performing verification of proofs for ProveCommits that are not valid.

Reply:
The new sector proof is valid even if the replaced sector is faulty. We just _won't remove the replaced sector_ if it's faulty.